text/0083-rawkv-change-data-capture.md

```diff
@@ -0,0 +1,152 @@
+# RawKV Change Data Capture
+
+- RFC PR: https://github.com/tikv/rfcs/pull/86
+- Tracking Issue: https://github.com/tikv/tikv/issues/11965
+
+## Summary
+
+This proposal introduces the technical design of **RawKV Change Data Capture** (abbr. **RawKV CDC**).
+
+## Motivation
+
+Customers are deploying TiKV clusters as non-transactional key-value (**RawKV**) storage for application router or other scenarios. When **RawKV** being more and more widely adopted, [Change Data Capture] become a necessary feature to meet the high availability requirement of industries (*e.g.* banking and finance, to deploy disaster recovery clusters).
+
+*(Now we will focus on the scenario of **Cross Cluster Replication**, i.e. replication between main and recovery TiKV clusters. Design for other scenarios will be supplemented later.)*
+
+In this proposal we introduces **TiKV-CDC**, a new component for **RawKV CDC**, which captures changed data from upstream/main TiKV cluster, and replicates to downstream/recovery TiKV cluster in real time.
+
+<img src="../media/rawkv-cdc.png" alt="RawKV CDC" width="60%"/>
+
+## Key Challenges & Solutions
+
+### Capturing Change
+
+Timestamp is necessary for indicating which and when data is changed. So **RawKV CDC** will base on [TiKV API V2], which encodes timestamp with key.
+
+### Consistency and Resolved Timestamp
+
+Data in downstream cluster is expected to be consistent with upstream cluster. To be consistent:
+
+**1st**, as **RawKV** does not provide transaction semantic, **RawKV CDC** will not guarantee the order of writes for different keys between upstream and downstream cluster.
+
+**2nd**, [TiKV API V2] provides causality consistence, that if two writes (of the same key) `A` [Happened Before] `B`, then `Timestamp(A)` < `Timestamp(B)`. **RawKV CDC** guarantee the same causality consistence as well.
+
+**3rd**, [TiKV API V2] does not guarantee that a **RawKV** entry with smaller timestamp must be observed first. So **TiKV-CDC** will sort entires by timestamp before send to downstream.
+
+But how **TiKV-CDC** knows it will not receive any even earlier entry from TiKV after the previous batch of data had been sent to downstream ? So we need **Resolved Timestamp** (abbr. **resolved-ts**). **Resolved-ts** is similar to [Watermark], by which TiKV indicating that all entries earlier have been observed. When **TiKV-CDC** receives a `resolved-ts`, all sorted entries with `timestamp <= resolved-ts` are safe to write to downstream.
```

Comment:
```suggestion
But how **TiKV-CDC** knows it will not receive any even earlier entry from TiKV after the previous batch of data had been sent to downstream ? So we need **Resolved Timestamp** (abbr. **resolved-ts**). **Resolved-ts** is similar to [Watermark], by which TiKV indicats that all entries earlier have been observed. When **TiKV-CDC** receives a `resolved-ts`, all sorted entries with `timestamp <= resolved-ts` are safe to write to downstream.
```